internal/expr/complete.go

```diff
@@ -0,0 +1,49 @@
+package expr
+
+import (
+	"fmt"
+	"reflect"
+)
+
+type CompletedExpr struct {
+	ParsedExpr  *ParsedExpr
+	InputValues []any
+}
+
+type typeNameToValue = map[string]any
+
+// Complete extracts query arguments specified in inputParts from structs.
```

Comment:
I misunderstood this as `inputParts` being extracted from structs, I think the terms need a little re-ordering, perhaps something like
```go
// Complete populates query arguments
```

Also the name 'Complete' seems a little vague to me? Perhaps it could be something like 'Populate' or 'Args'?

Reply:
Changed it make it more clear